restructure tools section (#211)
moj-analytical-services/user-guidance: source/documentation/tools/jupyterlab/package-management.md

# Python Package Management

## venv and pip

[pip](https://pip.pypa.io/en/stable/) is a terminal command used to install and upgrade Python packages.

[PyPI](https://pypi.org) is the main Python package repository. It's 'official', but that doesn't mean a lot - like most of these open source package repositories, a poor quality or even malicious package can easily be uploaded there, so do your diligence when picking them.

> A [Python virtual environment](https://docs.python.org/3/tutorial/venv.html) (or `venv`, for short) is a directory you can install a particular python executable and python packages into, away from your machine's default ones. Typically each project/repo you work on should have a different venv, and then you never have to deal with conflicting requirements between projects. When you 'activate' a particular venv, then when you run `python` or `pip`, then it will work with that venv's python executable and python packages.

Reviewer: ```suggestion
A [Python virtual environment](https://docs.python.org/3/tutorial/venv.html) (or `venv`, for short) is a directory you can install a particular Python executable and Python packages into, away from your machine's default ones. Typically each project/repo you work on should have a different venv, and then you never have to deal with conflicting requirements between projects. When you 'activate' a particular venv, then run `python` or `pip`, those commands will work with that venv's Python executable and Python packages.
```

> ## Basic usage

Reviewer: Not necessarily something for this pull request to solve, but there's some confusing repetition between the 'JupyterLab' page, which has a section on 'Using a virtual environment in Jupyter', and this section, which is part of a page called 'Python package management'. Currently the package management page is a directly child of the JupyterLab page.

I suspect the best option is to delete the virtual environment material from the JupyterLab page, as the stuff here on the package management page is more visible in the navigation.

```bash
(venv) $ python3 -c 'import pandas; print(pandas); print("It worked")'
<module 'pandas' from '/home/jovyan/myproject/venv/lib/python3.7/site-packages/pandas/__init__.py'>
It worked
```

In JupyterLab, to be able to use the venv's packages (instead of the system packages), see [Using a venv in Jupyter](index.html#using-a-virtual-environment-in-jupyter)

When you commit your code, to ensure reproducibility, you should also commit an up-to-date record of what packages you've installed. The simplest way is to do:

```bash
(venv) $ pip freeze >requirements.txt
(venv) $ git add requirements.txt
```

You should also add to your README file the instructions for using requirements.txt - see the following section.

## Using a project that has a requirements.txt

If a project has a 'requirements.txt' then you should install that into a venv.

A project's README file is the traditional place to communicate usage of a requirements.txt. Because of that, this section is provided in markdown format so it can be copied into your project's README, and tailored as necessary:

```markdown
## Setup

> Before you can run this project, you need some files setup in your home dir, using the terminal:

Reviewer: ```suggestion
Before you can run this project, you need to install some Python packages using the terminal:
```
